common/src/test/java/org/conscrypt/javax/crypto/CipherBasicsTest.java

```diff
@@ -85,6 +86,125 @@ public static void setUp() {
         TestUtils.assumeAllowsUnsignedCrypto();
     }
 
+    private enum CallPattern {
+        DO_FINAL,
+        DO_FINAL_WITH_OFFSET,
+        UPDATE_DO_FINAL,
+        MULTIPLE_UPDATE_DO_FINAL,
+        UPDATE_DO_FINAL_WITH_OUTPUT_ARRAY,
+        UPDATE_DO_FINAL_WITH_OUTPUT_ARRAY_AND_OFFSET,
+        DO_FINAL_WITH_INPUT_OUTPUT_ARRAY,
+        DO_FINAL_WITH_INPUT_OUTPUT_ARRAY_AND_OFFSET,
+        UPDATE_DO_FINAL_WITH_INPUT_OUTPUT_ARRAY
+    }
+
+    /** Concatenates the given arrays into a single array.*/
+    byte[] concatArrays(byte[]... arrays) {
```

Comment:
nit: We already have ArrayUtils.concat for the two array case.  Maybe move this method into ArrayUtils for the other cases (it's fairly small).